packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/links-section/link-table/use-row-data.ts

```diff
@@ -1,45 +1,50 @@
 import { useMemo } from "react";
+import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
+import {
+  getOutgoingLinkAndTargetEntitiesAtMoment,
+  getOutgoingLinksForEntityAtMoment,
+} from "@hashintel/hash-subgraph/src/stdlib/edge/link";
+import { getEntityTypeById } from "@hashintel/hash-subgraph/src/stdlib/element/entity-type";
 import { sortRowData } from "../../../../../../../components/GlideGlid/utils/sorting";
 import { generateEntityLabel } from "../../../../../../../lib/entities";
-import {
-  getPersistedEntityType,
-  getPersistedLinkType,
-} from "../../../../../../../lib/subgraph";
 import { useEntityEditor } from "../../entity-editor-context";
 import { LinkRow } from "./types";
 
 export const useRowData = () => {
-  const { rootEntityAndSubgraph, linkSort } = useEntityEditor();
+  const { entitySubgraph, linkSort } = useEntityEditor();
 
   const rowData = useMemo<LinkRow[]>(() => {
-    if (!rootEntityAndSubgraph) {
+    if (!entitySubgraph) {
       return [];
     }
 
-    const entity = rootEntityAndSubgraph.root;
-
-    return (
-      entity?.links.map((link) => {
-        const linkType = getPersistedLinkType(
-          rootEntityAndSubgraph.subgraph,
-          link.linkTypeId,
-        )?.inner;
-
-        const referencedEntityType = getPersistedEntityType(
-          rootEntityAndSubgraph.subgraph,
-          link.targetEntity.entityTypeId,
-        )?.inner;
-
-        return {
-          expectedEntityType: referencedEntityType?.title ?? "",
-          linkedWith: generateEntityLabel(rootEntityAndSubgraph),
-          linkId: link.linkTypeId,
-          relationShip: "Outbound",
-          type: linkType?.title ?? "",
-        };
-      }) ?? []
-    );
-  }, [rootEntityAndSubgraph]);
+    const entity = getRoots(entitySubgraph)[0]!;
+
+    return getOutgoingLinkAndTargetEntitiesAtMoment(
+      entitySubgraph,
+      entity.metadata.editionId.baseId,
+      /** @todo - We probably want to use entity endTime - https://app.asana.com/0/1201095311341924/1203331904553375/f */
+      new Date(),
+    ).map(({ linkEntity, rightEntity }) => {
+      const linkEntityType = getEntityTypeById(
+        entitySubgraph,
+        linkEntity.metadata.entityTypeId,
+      );
+
+      const referencedEntityType = getEntityTypeById(
+        entitySubgraph,
+        rightEntity.metadata.entityTypeId,
+      );
+
+      return {
+        expectedEntityType: referencedEntityType?.schema.title ?? "",
+        linkedWith: generateEntityLabel(entitySubgraph),
```

Comment:
Just making sure that we only have a single root in this `entitySubgraph` here. The subgraph doesn't have more roots, right?

Reply:
Correct, if there are more roots that would be treated as a bug. It's from the `EntityEditor` context which should have a subgraph rooted at the entity being edited

Reply:
I was going to suggest to the frontend devs that they try and encapsulate that invariant in the type system